apis/Google.Cloud.Firestore/Google.Cloud.Firestore/Query.cs

```diff
@@ -509,6 +511,50 @@ object ConvertValueToDocumentReferencesForInQuery()
             }
         }
 
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="filter"></param>
+        /// <returns></returns>
+        public Query Where(Filter filter)
+        {
+            var newFilters = _filters == null ? new List<InternalFilter>() : new List<InternalFilter>(_filters);
```

Comment:
The crux of my suggestion is that we get rid of the concept of having a list of filters in Query. Instead of the `_filters` field, we have a field which is just a single filter, *always* a `CompositeFilter` with a type of "and". An unfiltered query would have 0 elements and can be omitted; we can simplify a filter with 1 element to just that element; with more than 1 element we'd combine them in the normal way.